safe/impact_reports/area_exposure_report_mixin.py

```diff
@@ -0,0 +1,315 @@
+     the Free Software Foundation; either version 2 of the License, or
+     (at your option) any later version.
+"""
+__author__ = 'Samweli Twesa Mwakisambwe "Samweli" <smwltwesa6@gmail.com>'
+__date__ = '13/10/15'
+
+from collections import OrderedDict
+
+from safe.utilities.i18n import tr
+from safe.impact_reports.report_mixin_base import ReportMixin
+
+import safe.messaging as m
+
+
+class AreaExposureReportMixin(ReportMixin):
+    """Population specific report.
+    """
+
+    def __init__(self):
+        """Area specific report mixin.
+
+        .. versionadded:: 3.2
+
+        ..Notes::
+
+
+        """
+        self._question = ''
+        self._areas = {}
+        self._affected_areas = {}
+        self._areas_population = {}
+        self._total_population = 0
+        self._unaffected_population = 0
+        self._affected_population = {}
+        self._other_population_counts = {}
+
+    def generate_report(self):
+        """Breakdown by building type.
+
+        :returns: The report.
+        :rtype: list
+        """
+        message = m.Message()
+        message.add(m.Paragraph(self.question))
+        message.add(self.impact_summary())
+
+        return message
+
+    def impact_summary(self):
+        """The impact summary as per category
+
+        :returns: The impact summary.
+        :rtype: safe.messaging.Message
+        """
+        message = m.Message(style_class='container')
+        table = m.Table(style_class='table table-condensed table-striped')
+        table.caption = None
+
+        row = m.Row()
+        row.add(m.Cell(tr('Area id'), header=True))
+        row.add(m.Cell(tr('Affected Area (ha)'), header=True))
+        row.add(m.Cell(tr('Affected Area (%)'), header=True))
+        row.add(m.Cell(tr('Total (ha)'), header=True))
+        row.add(m.Cell(tr('Affected People'), header=True))
+        row.add(m.Cell(tr('Affected People(%)'), header=True))
+        row.add(m.Cell(tr('Total Number of People'), header=True))
+
+        table.add(row)
+
+        second_row = m.Row()
+        second_row.add(m.Cell(tr('All')))
+        total_affected_area = self.total_affected_areas
+        total_area = self.total_areas
+        percentage_affected_area = ((total_affected_area / total_area ) \
+                                    if total_area != 0 else 0) * 100
+        percentage_affected_area = round(percentage_affected_area, 1)
+
+        total_affected_population = self.total_affected_population
+        total_population = self.total_population
+        percentage_affected_people = ((total_affected_population / total_population ) \
+                                    if total_population != 0 else 0) * 100
+        percentage_affected_people = round(percentage_affected_people, 1)
+        total_affected_area *= 1e8
+        total_affected_area = round(total_affected_area, 0)
+        total_area *= 1e8
+        total_area = round(total_area, 0)
+
+        second_row.add(m.Cell(total_affected_area))
```

Comment:
Note that for all numeric cells we should have right alignment e.g.

```m.Cell(affected, align='right')```

The headers for numeric columns should also be right aligned